### ISSUE TITLE
[3.3.2 and earlier] In rare cases a child redactor field breaks, 400 on resave

### ISSUE THREAD

**Issue by maxstrebel** [spicywebau/craft-neo#642]
#### Description

In rare cases we experience a broken redactor field, when it is loaded in a matrix in a child block. It seems it might happen when you save while a draft is saved or merged. If you resave, you get a 400 and the error `No draft was identified by the request.`

#### Steps to reproduce
1. Create a matrix field that contains a redactor field
2. Create a Neo Block as parent
3. Create a Neo Child that contains the Matrix with the redactor field
4. Add elements and save
5. In some occasions, redactor will fail to init
6. Resave
7. Get 400 Error

#### Other information
- Neo version: 3.3.2 (but was already present in 3.2.5)
- Craft version: 4.2.1.1
- (if multi-site) What is the affected Neo field's propagation method:
- Is eager-loading used?

```
yii\web\BadRequestHttpException: No draft was identified by the request. in /app/vendor/craftcms/cms/src/controllers/ElementsController.php:1237
Stack trace:
#0 [internal function]: craft\controllers\ElementsController->actionApplyDraft()
#1 /app/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#2 /app/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#3 /app/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('apply-draft', Array)
#4 /app/vendor/craftcms/cms/src/web/Application.php(301): yii\base\Module->runAction('elements/apply-...', Array)
#5 /app/vendor/craftcms/cms/src/web/Application.php(625): craft\web\Application->runAction('elements/apply-...', Array)
#6 /app/vendor/craftcms/cms/src/web/Application.php(280): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#7 /app/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#8 /app/web/index.php(12): yii\base\Application->run()
#9 {main}
```

https://user-images.githubusercontent.com/910594/184974978-28f38db4-d5e5-434b-9483-d4e23b793056.mov

**Comment by maxstrebel** [comment]
I tested it with Matrix (only) and cannot reproduce it.

**Comment by ttempleton** [comment]
I was unable to reproduce the JavaScript error, strangely, since it concerns trying to access a Craft property that no longer exists in Craft 4. But the offending code was there to fix an error involving static Super Table sub-fields, which doesn't seem to be happening anymore when I remove the code (presumably related to the block rendering being rewritten for the Craft 4 version of Neo). So I've removed that code for the next release.

I assume the 400 error is a direct result of the JavaScript error, but if that still happens with the next release, please let me know.

**Comment by maxstrebel** [comment]
Thank you Thomas. So do you think I should report something to the redactor plugin?

**Comment by ttempleton** [comment]
No need for that, I assume that the JavaScript error coming from Neo was the only thing stopping the Redactor field from loading, and that the Redactor field should be working fine with the next Neo release (which should be in the next couple of days, hoping to fully resolve #639 first).

**Comment by maxstrebel** [comment]
Thank you. Appreciate your hard work.

**Comment by ttempleton** [comment]
I've just released Neo 3.3.3 with the fixes for this and #641. Hopefully that puts an end to both the broken Redactor field and the 400 error, but if either of those errors still occur, please let me know.
